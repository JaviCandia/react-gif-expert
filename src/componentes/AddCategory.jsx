import { useState } from "react";

export const AddCategory = ({ onAddCategory }) => {
  const [inputValue, setInputvalue] = useState("");

  const onInputChange = ({ target }) => {
    setInputvalue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    const trimmedValue = inputValue.trim();
    if (trimmedValue.length <= 1) return;

    // setCategories((categories) => [...categories, inputValue]); // using useState hook
    onAddCategory(trimmedValue);
    setInputvalue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input type="text" placeholder="Buscar Gifs" value={inputValue} onChange={onInputChange} />
    </form>
  );
};
