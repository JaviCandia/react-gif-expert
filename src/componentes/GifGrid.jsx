import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category, onDeleteCategory }) => {
  const { gifs, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3> {category} </h3>
      <button onClick={() => onDeleteCategory(category)} style={{color: 'white', background: 'red', border: 'none'}}>X</button>
      {isLoading && <h2> Cargando...</h2>}
      <div className="card-grid">
        {gifs.map((gif) => {
          return <GifItem key={gif.id} {...gif} />; // {...object } propagates all props to child
        })}
      </div>
    </>
  );
};
