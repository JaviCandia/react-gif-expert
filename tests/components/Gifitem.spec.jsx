import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/componentes/GifItem";
import { renderIntoDocument } from "react-dom/test-utils";

describe("GifItem", () => {
  const title = "Jujutsu";
  const url = "https://www.gifs.com/jotaro";

  it("should match Snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  it("should should image with correct url and alt", () => {
    render(<GifItem title={title} url={url} />);

    // expect(screen.getByTestId("gif-image").src).toBe(url);
    const { src, alt } = screen.getByTestId("gif-image");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  it("should show title in the component", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByTestId("title").innerHTML).toBe(title);
  });
});
