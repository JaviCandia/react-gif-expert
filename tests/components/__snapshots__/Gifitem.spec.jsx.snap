// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem should match Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Jujutsu"
      data-testid="gif-image"
      src="https://www.gifs.com/jotaro"
    />
    <p
      data-testid="title"
    >
      Jujutsu
    </p>
  </div>
</div>
`;
